lte/gateway/c/session_manager/ShardTracker.h

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright 2020 The Magma Authors.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <string>
+#include <utility>
+#include <vector>
+#include <map>
+
+namespace magma {
+
+/*Shards represent groups of UEs placed into buckets of
+a certain size, to make polling more manageable*/
+class ShardTracker {
+  ShardTracker();
+
+  /**
+   * add UE to shards based on availability
+   * @return index(shard id) where UE was placed
+   */
+  // TODO(veshkemburu): Store IMSI as well for easier subscriber reallocation
+  // (GH8167)
+  int add_ue();
+
+  /**
+   * remove UE from shard
+   * @param shard_id location of UE to be removed
+   */
+  void remove_ue(int shard_id);
+
+ private:
+  // a vector of quantities, where the indices represent
+  // the shard id and the values represent the number of
+  // UEs held in each shard
+  std::vector<uint16_t> shards_;
+  // largest number of UEs that can fill a shard
+  uint16_t max_shard_size_;
+  // the number of shards we currently have
+  uint16_t number_of_shards_;
```

Comment:
Is `number_of_shards_` always equal to `shards_.size()`? Or do we want this number to omit  some of 0 values in the shards_ vector? 

If we just want the number of shards that exist, maybe we can just use the size of shards_.
If not, we should clarify it to be like `number_of_shards_with_ues_`?

Reply:
It was just meant to be the shards size, but it isn't necessary, you're right.